File: api/v1/schemas/market_sync.py
# -*- coding: utf-8 -*-
"""Market data sync API schemas."""

from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel, Field


class MarketSyncRunRequest(BaseModel):
    markets: List[str] = Field(default_factory=list, description="Markets to sync: cn/us")


class MarketSyncStatusResponse(BaseModel):
    running: bool
    started_at: Optional[str] = None
    finished_at: Optional[str] = None
    current_market: Optional[str] = None
    current_code: Optional[str] = None
    processed: int
    saved: int
    skipped: int
    errors: int
    total_candidates: int
    priority_candidates: int
    priority_processed: int
    priority_completed: int
    message: str
    markets: List[str] = Field(default_factory=list)


class MarketSyncRunResponse(BaseModel):
    accepted: bool
    message: str
    status: MarketSyncStatusResponse

File: apps/dsa-web/src/api/marketSync.ts
import apiClient from './index';
import { toCamelCase } from './utils';

export interface MarketSyncStatus {
  running: boolean;
  startedAt?: string | null;
  finishedAt?: string | null;
  currentMarket?: string | null;
  currentCode?: string | null;
  processed: number;
  saved: number;
  skipped: number;
  errors: number;
  totalCandidates: number;
  priorityCandidates: number;
  priorityProcessed: number;
  priorityCompleted: number;
  message: string;
  markets: string[];
}

export interface MarketSyncRunResponse {
  accepted: boolean;
  message: string;
  status: MarketSyncStatus;
}

export const marketSyncApi = {
  async getStatus(): Promise<MarketSyncStatus> {
    const response = await apiClient.get<Record<string, unknown>>('/api/v1/system/market-sync/status');
    return toCamelCase<MarketSyncStatus>(response.data);
  },

  async run(markets: Array<'cn' | 'us'>): Promise<MarketSyncRunResponse> {
    const response = await apiClient.post<Record<string, unknown>>('/api/v1/system/market-sync/run', {
      markets,
    });
    return toCamelCase<MarketSyncRunResponse>(response.data);
  },
};
